api/grpcserver/v2alpha1/malfeasance_test.go

```diff
@@ -370,10 +370,8 @@ func TestMalfeasanceStreamService_Stream(t *testing.T) {
 		}
 		stream, err := client.Stream(context.Background(), request)
 		require.NoError(t, err)
-		md, err := stream.Header()
-		require.NoError(t, err)
-		require.Equal(t, []string{"application/grpc"}, md.Get("content-type"))
 
+		time.Sleep(100 * time.Millisecond) // wait for the stream handler to subscribe to events before sending them
```

Comment:
This test will be flaky on a slower machine. What if it takes > 100ms to register for events?

Reply:
What do you suggest? Make the sleep 1 second?

Reply:
I suggest changing the grpc service logic back to first registering for events and then reading the DB so that it is guaranteed that proofs aren't lost due to races and implement the simple deduplication logic I suggested earlier.

The test could:
1. put N proofs in the DB
2. start the proofs GRPC stream
3. read N proofs from the stream
4. report M proofs
5. read M proofs from the stream

This test wouldn't be flaky and there is no need for the synchronization via header.